File: src/main/java/kr/co/lunatalk/infra/jwt/JwtProperties.java
package kr.co.lunatalk.infra.jwt;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "jwt")
public record JwtProperties(
	String accessTokenSecret,
	String refreshTokenSecret,
	Long accessTokenExpirationTime,
	Long refreshTokenExpirationTime,
	String issuer
) {
	public Long getAccessTokenExpirationTime() {
		return accessTokenExpirationTime * 1000;
	}
	public Long getRefreshTokenExpirationTime() {
		return refreshTokenExpirationTime * 1000;
	}
}

File: lunatalk-domain/src/main/kotlin/kr/co/lunatalk/infra/config/jwt/JwtProperties.kt
package kr.co.lunatalk.infra.config.jwt

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "jwt")
data class JwtProperties(
    val accessTokenSecret: String,
    val refreshTokenSecret: String,
    val accessTokenExpirationTime: Long,
    val refreshTokenExpirationTime: Long,
    val issuer: String
) {
    fun accessTokenExpirationTimeMillis(): Long = accessTokenExpirationTime * 1000

    fun refreshTokenExpirationTimeMillis(): Long = refreshTokenExpirationTime * 1000
}
